src/pages/ReservationDetails.tsx

```diff
@@ -13,146 +26,259 @@ export default function ReservationApproval() {
 
-  // TODO: Load context from server!
-  const reservation: Reservation = {};
+  const [reservation, setReservation] = useState<Reservation>();
+  const [error, setError] = useState<string>();
+  const [confirmationButtonText, setConfirmationButtonText] = useState<string>(
+    'Confirm Reservation',
+  );
+  const [cancelButtonText, setCancelButtonText] =
+    useState<string>('Cancel Reservation');
+
+  const { reservationApi } = useContext(Context);
+  useEffect(() => {
+    reservationApi
+      .getReservation(reservationId ?? '0')
+      .then(val => {
+        setReservation(val);
+      })
+      .catch(e => {
+        if (e instanceof Response) {
+          e.json().then(j =>
+            (j.message as string).startsWith('Unable to find')
+              ? setError('Reservation not found!')
+              : setError(''),
+          );
+        }
+      });
+  }, []);
 
-  /**
-   * Checks if the given timestamp is tommorrow
-   * @param {number} timestamp the unix timestamp from reservation.time.from
-   * @return {boolean}
-   */
-  function isIn12Hours(timestamp: number): boolean {
+  const isIn12Hours = (timestamp: number): boolean => {
     const now = new Date();
     const difference = (timestamp - now.getTime()) * 60 * 60 * 1000;
     if (difference > 0 && difference <= 12) {
       return true;
     }
     return false;
-  }
+  };
+
+  const cancelReservation = async () => {
+    setCancelButtonText('...');
+    window.confirm('Are you sure you want to cancel your reservation?') &&
+      reservationApi
+        .deleteReservation(reservationId ?? '')
+        .then(() => {
+          setCancelButtonText('Reservation canceled!');
+        })
+        .catch(() => {
+          setCancelButtonText('Something went wrong!');
+          setError('Something went wrong canceling your reservation');
+        });
+  };
+
+  const confirmReservation = async () => {
+    setConfirmationButtonText('...');
+    await reservationApi
+      .putReservation(
+        { confirmed: true },
+        reservation?.id ?? '',
+        confirmationToken ?? '',
+      )
+      .then(() => {
+        setConfirmationButtonText('reservation confirmed!');
+        reservation !== undefined ? (reservation.confirmed = true) : null;
+      })
+      .catch(() => {
+        setConfirmationButtonText('Something went wrong!');
+        setError('Something went wrong confirming your reservation');
+      });
+  };
 
   return (
-    <div className='booking-summary-container'>
-      <h3>Reservation summary</h3>
-      <div className='booking-summary-date_person-container'>
-        <div>
-          <p className='booking-summary-label'>Date</p>
-          <DatePicker
-            value={reservation.time?.from ?? 0}
-            onChange={() => {
-              return;
-            }}
-            renderInput={params => (
-              <TextField {...params} size='small' fullWidth disabled />
+    <>
+      {error ? (
```

Comment:
Mach den alert bitte in ein `Collapse` rein, damit es animiert und konsistent ist, siehe hier:
https://github.com/Eistbaren/client/blob/69d62231f1a41228bb9a525c2820fa99b23fa733/src/pages/PersonalData.tsx#L84